tests/unit/features/test_features_models.py

```diff
@@ -24,6 +26,99 @@ def test_initializing_multivariate_feature_state_value_creates_default_uuid():
     assert mv_fs_value_model.mv_fs_value_uuid is not None
 
 
+def test_feature_state_model__invalid_multivariate_feature_state_values__raises_expected(
+    feature_1: FeatureModel,
+) -> None:
+    # Given
+    invalid_multivariate_feature_state_values = [
+        MultivariateFeatureStateValueModel(
+            multivariate_feature_option=MultivariateFeatureOptionModel(value="test"),
+            percentage_allocation=50,
+        ),
+        MultivariateFeatureStateValueModel(
+            multivariate_feature_option=MultivariateFeatureOptionModel(value="test2"),
+            percentage_allocation=51,
+        ),
+    ]
+
+    # When
+    with pytest.raises(ValidationError) as exc:
+        FeatureStateModel(
+            feature=feature_1,
+            multivariate_feature_state_values=invalid_multivariate_feature_state_values,
+            enabled=True,
+        )
+
+    # Then
+    [error_data] = exc.value.errors()
+    assert error_data["loc"] == ("multivariate_feature_state_values",)
+    assert error_data["type"] == "value_error.invalidpercentageallocation"
```

Comment:
Should there be a `"text"` or `"body"` key here as well that we can check?

Reply:
Do you mean `msg`? The message text is tested in `test_feature_state_model__multivariate_feature_state_values__append_invalid__raises_expected`.

Reply:
Agreed, done.